code/datums/components/technointrovert.dm

```diff
@@ -0,0 +1,36 @@
+/// You can't use machines unless they are in a whitelist
+/datum/component/technointrovert
+	can_transfer = TRUE
+	/// Typecache of allowed machinery
+	var/list/whitelist
+	/// Message presented when
+	var/message = "That %TARGET is strange! Let's avoid it."
+
+/datum/component/technointrovert/Initialize(_whitelist, _message)
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	whitelist = _whitelist
+	if(_message)
+		message = _message
+
+/datum/component/technointrovert/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_CAN_USE_MACHINE, .proc/is_in_friend_group)
```

Comment:
does this also prevent them from just bashing a machine or using one of their tools to interact with it?

can drones crowbar open doors, or will they have to ventcrawl around them?

can drones create machine frames or computer frames that they then can't interact with/deconstruct/bash to pieces?

Reply:
> does this also prevent them from just bashing a machine or using one of their tools to interact with it?

This just applies to using the machine, like with your hands

> can drones crowbar open doors, or will they have to ventcrawl around them?

Doors don't have the INTERACT_ATOM_ATTACK_HAND flag, so they ignore the can_interact proc completely

> can drones create machine frames or computer frames that they then can't interact with/deconstruct/bash to pieces?

machine frames and computer frames are not /obj/machinery, however I forgot computer circuitboards in the whitelist, thanks